### ISSUE TITLE
Make the output report table a rspec/serverspec output format

### ISSUE THREAD

**Issue by mcallaway** [vincentbernat/serverspec-example#10]
It is already possible to use an "output formatter", like --format html, to product an HTML output report. Would it be possible to convert your angular JS report table into such an output format, to remove the need for a web server/viewer?

I imaging a Jenkins job using the HTML Publisher that would run:

rake -m -j 5 check:server:all

and the table would be the final report.

**Comment by vincentbernat** [comment]
It surely possible to embed the JSON file into an HTML file, open the HTML file and get the report this way. However, I am unsure if that's what you would expect when you say "HTML Publisher".

**Comment by mcallaway** [comment]
What I want is as follows. Right now I can do this:

    -> cat > .rspec <EOF
    --color
    --format documentation
    --format html
    --out report.html
    EOF

Then:

    bundle exec rake -m -j 5 check:server:all

I then see normal "documentation" output on my terminal, and I get HTML output in report.html. What I want is for your fancy tabular output to be an option as well, for example an output named "serverspec":

    -> cat > .rspec <EOF
    --color
    --format documentation
    --format serverspec
    --out report.html

Such that when I have Jenkins run the tests:

    bundle exec rake -m -j 5 check:server:all

I could then view the results via the jenkins HTML Publisher plugin, which essentially just exposes an HTML file via the Jenkins UI.

In short, I just want to be able to "open report.html" and see the results, without having to run "rake viewer". Right now I'm having a hard time getting Jenkins to show the report, due to "security" features (external js, css, the call to "url" or "file" route).

**Comment by vincentbernat** [comment]
I understand. Unfortunately, I currently don't have time to implement this myself. Inside `index.html`, I would put the JSON file into a `<script>` tag:

```html
<script>
__preloaded = {...}
</script>
```

I would open `report-XXXX.html#/preloaded`.

In `report.js`, I would add a route:

```js
when("/preloaded", {
  templateUrl: "result.html",
  controller: "reportResultCtrl",
  resolve: {
    filename: "preloaded",
    data: __preloaded
  }
})
```

Maybe the `resolve` stuff should be enclosed in functions. My Angular is a bit rusty.

Note that you still have to have a webserver able to serve the CSS and the JS in the appropriate locations. Otherwise, you would need to embed everything in the HTML. If you can arrange for CSS and JS to be at the right location, maybe it would be as easy open index.html like this:

 http://jenkins.internal/serverspec/viewer/index.html#/url/http://jenkins.internal/jobs/serverspec/artifacts/report-XXXX.json

**Comment by mcallaway** [comment]
Thanks for the tip. I'll try it. So far I get this rather unhelpful:

    angular.js:9402Error: [$injector:unpr] http://errors.angularjs.org/1.2.7/$injector/unpr?p0=preloadedProvider%20%3C-%20preloaded
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:6:454
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:32:127
    c@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:30:201
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:32:196
    c@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:30:201
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular-route.min.js:10:296
    q@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:7:384
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular-route.min.js:10:255
    C@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:92:424
    C@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:92:424
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:94:65
    $eval@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:102:293
    $digest@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:100:55
    $apply@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:103:107
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:18:29
    d@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:30:457
    c@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:17:445
    Yb@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:18:141
    Sc@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:17:216
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:202:25
    a@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:131:354
    https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:27:212
    q@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:7:384
    c@https://ajax.googleapis.com/ajax/libs/angularjs/1.2.7/angular.min.js:27:191

**Comment by mcallaway** [comment]
I think I got it to work:

    50         when("/preloaded", {
    51             templateUrl: "result.html",
    52             controller: "reportResultCtrl",
    53             resolve: {
    54               filename: ["$route", function($route) {
    55                   return "preloaded";
    56               }],
    57                 data: ["$route", "ResultData", function($route, ResultData) {
    58                   return __preloaded;
    59               }]
    60             }
    61         }).
